src/hb-ot-color-sbix-table.hh

```diff
@@ -58,10 +64,17 @@ struct SBIXStrike
   inline bool sanitize (hb_sanitize_context_t *c) const
   {
     TRACE_SANITIZE (this);
-    return_trace (c->check_struct (this) &&
+    if (!(c->check_struct (this) &&
 		  c->check_array (imageOffsetsZ,
 				  sizeof (HBUINT32),
-				  1 + c->num_glyphs));
+				  1 + c->num_glyphs)))
+      return_trace (false);
+
+    for (unsigned int i = 0; i < 1 + c->num_glyphs; ++i)
+      if (!c->check_range (this, imageOffsetsZ[i]))
+        return_trace (false);
```

Comment:
How about first finding the maximum imageOffsetsZ value then checking if that is in range.

Reply:
How is that not O(num-glyphs)? ;)